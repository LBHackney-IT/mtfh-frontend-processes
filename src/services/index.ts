export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
