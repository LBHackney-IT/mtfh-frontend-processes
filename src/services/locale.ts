import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
