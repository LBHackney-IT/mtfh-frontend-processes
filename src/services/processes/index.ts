export * from "./sole-to-joint";
export * from "./change-of-name";
