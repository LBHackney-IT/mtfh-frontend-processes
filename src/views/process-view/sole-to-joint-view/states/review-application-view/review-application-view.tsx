import React, { useState } from "react";

import { CloseProcessView } from "../../../../../components";
import { locale } from "../../../../../services";
import { IProcess } from "../../../../../types";
import { isCurrentState } from "../../../../../utils/processUtil";
import { DesBox } from "../../../shared/process-components";
import { SubmitCaseView } from "../../../shared/submit-case-view";
import { HoReviewFailedView } from "../ho-review-view/ho-review-failed-view";
import { HoReviewView } from "../ho-review-view/ho-review-view";
import { NewTenancyView } from "../new-tenancy-view/new-tenancy-view";
import {
  EligibilityChecksPassedBox,
  TenureInvestigationRecommendationBox,
} from "../shared";
import { TenureInvestigationView } from "../tenure-investigation-view";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Box,
  Center,
  ErrorSummary,
  Spinner,
  StatusErrorSummary,
  StatusHeading,
} from "@mtfh/common/lib/components";

const { views } = locale;

interface ReviewApplicationViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
}

export const ReviewApplicationView = ({
  processConfig,
  process,
  mutate,
  optional,
}: ReviewApplicationViewProps): JSX.Element => {
  const [globalError, setGlobalError] = useState<number>();
  const [documentsSigned, setDocumentsSigned] = useState<boolean>(false);
  const { submitted, closeCase, setCloseCase } = optional;
  const { data: tenure, error } = useTenure(process.targetId);
  const {
    applicationSubmitted,
    tenureInvestigationFailed,
    tenureInvestigationPassed,
    tenureInvestigationPassedWithInt,
    hoApprovalPassed,
    hoApprovalFailed,
    tenureAppointmentScheduled,
    tenureAppointmentRescheduled,
    interviewScheduled,
    interviewRescheduled,
    tenureUpdated,
    processClosed,
  } = processConfig.states;

  if (error) {
    return (
      <ErrorSummary
        id="tenure-investigation-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  if (applicationSubmitted.state === process.currentState.state && submitted) {
    return (
      <SubmitCaseView
        processConfig={processConfig}
        process={process}
        mutate={mutate}
        optional={optional}
      />
    );
  }

  return (
    <div data-testid="soletojoint-ReviewApplication">
      {globalError && (
        <StatusErrorSummary id="review-application-global-error" code={globalError} />
      )}
      <EligibilityChecksPassedBox />
      <DesBox
        title={views.submitCase.supportingDocumentsApproved}
        description={views.submitCase.viewDocumentsOnDes}
      />

      {!isCurrentState(applicationSubmitted.state, process) && (
        <TenureInvestigationRecommendationBox
          processConfig={processConfig}
          process={process}
        />
      )}

      {applicationSubmitted.state === process.currentState.state && (
        <TenureInvestigationView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {[
        tenureInvestigationFailed.state,
        tenureInvestigationPassed.state,
        tenureInvestigationPassedWithInt.state,
        interviewScheduled.state,
        interviewRescheduled.state,
      ].includes(process.currentState.state) && (
        <HoReviewView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {(isCurrentState(hoApprovalFailed.state, process) ||
        isCurrentState(processClosed.state, process)) && (
        <HoReviewFailedView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
        />
      )}

      {[
        hoApprovalPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
        tenureUpdated.state,
      ].includes(process.currentState.state) && (
        <NewTenancyView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{
            closeCase,
            setCloseCase,
            documentsSigned,
            setDocumentsSigned,
            tenant,
          }}
        />
      )}

      {closeCase && (
        <>
          <Box variant="warning">
            <StatusHeading
              variant="warning"
              title={views.closeProcess.soleToJointClosed}
            />
          </Box>
          <CloseProcessView
            process={process}
            processConfig={processConfig}
            mutate={mutate}
            setGlobalError={setGlobalError}
          />
        </>
      )}
    </div>
  );
};
