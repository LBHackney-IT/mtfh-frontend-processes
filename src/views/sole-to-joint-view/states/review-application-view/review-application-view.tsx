import { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { SoleToJointHeader } from "../../../../components";
import { locale } from "../../../../services";
import { IProcess } from "../../../../types";
import { CloseProcessView } from "../../shared/close-process-view";
import { HoReviewFailedView } from "../ho-review-view/ho-review-failed-view";
import { HoReviewView } from "../ho-review-view/ho-review-view";
import { NewTenancyView } from "../new-tenancy-view/new-tenancy-view";
import { DesBox, EligibilityChecksPassedBox, TenantContactDetails } from "../shared";
import { TenureInvestigationView } from "../tenure-investigation-view";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Box,
  Center,
  ErrorSummary,
  Link,
  Spinner,
  StatusErrorSummary,
  StatusHeading,
  Text,
} from "@mtfh/common/lib/components";
import { BoxVariant } from "@mtfh/common/lib/components/box";
import { StatusHeadingVariant } from "@mtfh/common/lib/components/status-heading";

const { views } = locale;

interface ReviewApplicationViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
}
type Recommendation = "approve" | "decline" | "int";
const isCurrentState = (state, process) => state === process.currentState.state;
const isPreviousState = (state, process) =>
  process.previousStates.find((previousState) => state === previousState.state);

const getRecommendation = (
  processConfig: IProcess,
  process: Process,
): {
  recommendation: Recommendation;
  recommendationBoxVariant: BoxVariant;
  recommendationHeadingVariant: StatusHeadingVariant;
} => {
  const { tenureInvestigationFailed, tenureInvestigationPassed } = processConfig.states;
  if (
    isCurrentState(tenureInvestigationFailed.state, process) ||
    isPreviousState(tenureInvestigationFailed.state, process)
  ) {
    return {
      recommendation: "decline",
      recommendationBoxVariant: "warning",
      recommendationHeadingVariant: "warning",
    };
  }
  if (
    isCurrentState(tenureInvestigationPassed.state, process) ||
    isPreviousState(tenureInvestigationPassed.state, process)
  ) {
    return {
      recommendation: "approve",
      recommendationBoxVariant: "success",
      recommendationHeadingVariant: "success",
    };
  }
  return {
    recommendation: "int",
    recommendationBoxVariant: undefined,
    recommendationHeadingVariant: "base",
  };
};

export const ReviewApplicationView = ({
  processConfig,
  process,
  mutate,
  optional,
}: ReviewApplicationViewProps): JSX.Element => {
  const [globalError, setGlobalError] = useState<number>();
  const { closeCase, setCloseCase } = optional;
  const { data: tenure, error } = useTenure(process.targetId);
  const {
    applicationSubmitted,
    tenureInvestigationFailed,
    tenureInvestigationPassed,
    tenureInvestigationPassedWithInt,
    hoApprovalPassed,
    hoApprovalFailed,
    tenureAppointmentScheduled,
    tenureAppointmentRescheduled,
    interviewScheduled,
    interviewRescheduled,
    tenureUpdated,
    processClosed,
  } = processConfig.states;
  const {
    recommendation,
    recommendationBoxVariant,
    recommendationHeadingVariant,
  }: {
    recommendation: Recommendation;
    recommendationBoxVariant: BoxVariant;
    recommendationHeadingVariant: StatusHeadingVariant;
  } = getRecommendation(processConfig, process);

  if (error) {
    return (
      <ErrorSummary
        id="tenure-investigation-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  return (
    <div data-testid="soletojoint-ReviewApplication">
      <SoleToJointHeader processConfig={processConfig} process={process} />
      {globalError && (
        <StatusErrorSummary id="review-application-global-error" code={globalError} />
      )}
      <EligibilityChecksPassedBox />
      <DesBox title={views.submitCase.supportingDocumentsApproved} />

      {!isCurrentState(applicationSubmitted.state, process) && (
        <Box variant={recommendationBoxVariant}>
          <StatusHeading
            variant={recommendationHeadingVariant}
            title={views.tenureInvestigation.tenureInvestigatorRecommendation(
              recommendation,
            )}
          />
        </Box>
      )}

      {!closeCase &&
        ![
          tenureInvestigationFailed.state,
          tenureInvestigationPassed.state,
          tenureInvestigationPassedWithInt.state,
          interviewScheduled.state,
          interviewRescheduled.state,
          processClosed.state,
        ].includes(process.currentState.state) &&
        (tenant ? (
          <TenantContactDetails tenant={tenant} />
        ) : (
          <Text>Tenant not found.</Text>
        ))}

      {applicationSubmitted.state === process.currentState.state && (
        <TenureInvestigationView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
        />
      )}

      {[
        tenureInvestigationFailed.state,
        tenureInvestigationPassed.state,
        tenureInvestigationPassedWithInt.state,
        interviewScheduled.state,
        interviewRescheduled.state,
      ].includes(process.currentState.state) && (
        <HoReviewView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {(isCurrentState(hoApprovalFailed.state, process) ||
        isCurrentState(processClosed.state, process)) && (
        <HoReviewFailedView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
        />
      )}

      {[
        hoApprovalPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
      ].includes(process.currentState.state) && (
        <NewTenancyView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ closeCase, setCloseCase }}
        />
      )}

      {isCurrentState(tenureUpdated.state, process) && (
        <Box variant="success">
          <StatusHeading
            variant="success"
            title={views.tenureInvestigation.tenancySigned}
          />
          <div
            style={{ marginLeft: 60, marginTop: 17.5 }}
            className="govuk-link lbh-link lbh-link--no-visited-state"
          >
            <Link as={RouterLink} to="#" variant="link">
              {views.tenureInvestigation.viewNewTenure}
            </Link>
          </div>
        </Box>
      )}

      {(closeCase || isCurrentState(tenureUpdated.state, process)) && (
        <>
          <Box variant="warning">
            <StatusHeading variant="warning" title={views.closeCase.soleToJointClosed} />
          </Box>
          <CloseProcessView
            process={process}
            processConfig={processConfig}
            mutate={mutate}
          />
        </>
      )}
    </div>
  );
};
