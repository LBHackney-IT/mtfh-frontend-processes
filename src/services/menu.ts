import * as processes from "./processes";

import { Asset } from "@mtfh/common/lib/api/asset/v1/types";
import { Tenure } from "@mtfh/common/lib/api/tenure/v1/types";

interface MenuProps {
  label: string;
  link: string;
  processes?: { label: string; link: string; targetType: string }[];
  getPplQuery?: (context: any) => string;
}

const getProcessLink = (processName: string) => {
  const process = processes[processName];
  return {
    targetType: process.targetType,
    label: process.title,
    link: `/processes/${process.processName}/start`,
  };
};

const getPplQueryForRequestALegalReferral = ({ tenure, asset }): string => {
  if (!tenure || !asset) {
    return "";
  }
  const {
    householdMembers,
    tenuredAsset: { propertyReference },
  } = tenure as Tenure;
  const {
    assetAddress: { addressLine1, postCode },
  } = asset as Asset;
  const tenantNames = householdMembers
    .filter((member) => member.isResponsible)
    .map((member) => member.fullName)
    .join(", ");
  const address = addressLine1;
  const query = `usp=pp_url&entry.1699199605=${tenantNames}&entry.2086190845=${address}&entry.1956036523=${postCode}&entry.1140527432=${propertyReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const getPplQueryTenancyChange = ({ tenure }): string => {
  if (!tenure) {
    return "";
  }
  const {
    householdMembers,
    paymentReference,
    tenuredAsset: { fullAddress, propertyReference },
  } = tenure as Tenure;

  const tenant = householdMembers.find((member) => member.isResponsible);
  const query = `usp=pp_url&entry.1942863541=${tenant?.fullName}&entry.1090481808=${fullAddress}&entry.321634337=${propertyReference}&entry.582205322=${paymentReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const menu: MenuProps[] = [
  {
    label: "Sole to Joint",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    processes: [
      getProcessLink(processes.soletojoint.processName),
      getProcessLink(processes.changeOfName.processName),
    ],
  },
  {
    label: "Other tenancy changes",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    getPplQuery: getPplQueryTenancyChange,
  },
  {
    label: "Record a tenancy / audit homecheck",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform",
  },
  {
    label: "Find legal referral response",
    link: "https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140",
  },
  {
    label: "Request a legal referral",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform",
    getPplQuery: getPplQueryForRequestALegalReferral,
  },
  {
    label: "Record notice served",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform",
  },
  {
    label: "Rehousing",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform",
  },
  {
    label: "Person",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform",
  },
  {
    label: "Report Japanese knotweed",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628",
  },
  {
    label: "Report overgrown tree",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628",
  },
  {
    label: "Report overgrown garden",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628",
  },
];

export default menu;
