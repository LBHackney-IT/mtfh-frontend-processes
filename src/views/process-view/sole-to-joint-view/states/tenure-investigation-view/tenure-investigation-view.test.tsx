import {
  getContactDetailsV2,
  mockContactDetailsV2,
  mockProcessV1,
  patchProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen, waitForElementToBeRemoved } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale, processes } from "../../../../../services";
import { ReviewApplicationView } from "../review-application-view/review-application-view";

let submitted = false;
let closeCase = false;
const setSubmitted = () => {};
const setCloseCase = () => {};

describe("tenure-investigation-view", () => {
  beforeEach(() => {
    jest.resetModules();
    submitted = false;
    closeCase = false;
  });

  test("it renders ReviewApplication view correctly for ApplicationSubmitted state", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await expect(
      screen.findByText(locale.views.tenureInvestigation.tenureInvestigationCompleted, {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("it enables buttons when checkbox is checked", async () => {
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );

    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));

    const approve = screen.getByText(locale.views.tenureInvestigation.approve);
    const appointment = screen.getByText(locale.views.tenureInvestigation.appointment);
    const decline = screen.getByText(locale.views.tenureInvestigation.decline);
    await userEvent.click(
      screen.getByLabelText(
        locale.views.tenureInvestigation.tenureInvestigationCompleted,
      ),
    );
    expect(approve).toBeEnabled();
    expect(appointment).toBeEnabled();
    expect(decline).toBeEnabled();
    await userEvent.click(approve);
    await userEvent.click(appointment);
    await userEvent.click(decline);
  });

  test("it renders error if submit fails", async () => {
    server.use(patchProcessV1(null, 500));
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await userEvent.click(
      screen.getByLabelText(
        locale.views.tenureInvestigation.tenureInvestigationCompleted,
      ),
    );
    await userEvent.click(screen.getByText(locale.views.tenureInvestigation.approve));
    await expect(
      screen.findByText("There was a problem with completing the action", {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
  });
});
