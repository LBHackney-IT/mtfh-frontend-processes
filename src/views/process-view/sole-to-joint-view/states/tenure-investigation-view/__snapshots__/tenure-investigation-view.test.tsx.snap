// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tenure-investigation-view it renders ReviewApplication view correctly for ApplicationSubmitted state 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewApplication"
  >
    <h2
      class="lbh-heading-h2 entity-summary__tenure-heading"
    >
      <div>
        Tenure payment ref
         
        mt0xzvjp6g
      </div>
      <div>
        72251 Champlin Trafficway, 99647
      </div>
      <div>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/person/a302fe28-e5c3-4c84-a3e6-cd575e65093f"
          target="_blank"
        >
          Global Web Consultant Alverta Denesik
        </a>
      </div>
    </h2>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Supporting documents approved
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="http://des/teams/1/dashboard/requests"
          target="_blank"
        >
          View documents on the Document Evidence Store
        </a>
      </div>
    </div>
    <p
      class="lbh-body-s"
    >
      Global Web Consultant Alverta Denesik
       contact details:
    </p>
    <p
      class="lbh-body-s"
    >
      Phone:
      <span
        style="margin-left: 1em;"
      >
        075949281
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      Email:
      <span
        style="margin-left: 1em;"
      >
        Gay.Jacobi@hotmail.com
      </span>
    </p>
    <div
      class="govuk-checkboxes__item"
    >
      <input
        class="govuk-checkboxes__input"
        id="tenure-investigation-completed"
        type="checkbox"
      />
      <label
        class="govuk-label govuk-checkboxes__label"
        for="tenure-investigation-completed"
      >
        I confirm that the tenure investigation has been completed
      </label>
    </div>
    <p
      class="lbh-body-m"
    >
      Recommendation:
    </p>
    <button
      class="govuk-button lbh-button"
      disabled=""
      style="width: 222px; margin-right: 100%;"
      type="submit"
    >
      Approve
    </button>
    <button
      class="govuk-button lbh-button"
      disabled=""
      style="width: 222px; margin-right: 100%;"
      type="submit"
    >
      Appointment
    </button>
    <button
      class="govuk-button lbh-button"
      disabled=""
      style="width: 222px; margin-right: 100%;"
      type="submit"
    >
      Decline
    </button>
  </div>
</div>
`;
